src/app/query/bitcoin/transaction/use-bitcoin-broadcast-transaction.ts

```diff
@@ -19,20 +25,29 @@ export function useBitcoinBroadcastTransaction() {
   const client = useBitcoinClient();
   const [isBroadcasting, setIsBroadcasting] = useState(false);
   const analytics = useAnalytics();
+  const { checkIfUtxosListIncludesInscribed } = useCheckInscribedUtxos();
 
   const broadcastTx = useCallback(
     async ({
       tx,
       onSuccess,
       onError,
       onFinally,
+      sendingInscriptionsList = [],
       delayTime = 700,
-      checkForInscribedUtxos,
     }: BroadcastCallbackArgs) => {
       try {
-        // To-Do implement check in all btc related flows and remove conditional usage
-        // add explicit check in broadcastTx to ensure that utxos are checked before broadcasting
-        const hasInscribedUtxos = await checkForInscribedUtxos?.();
+        let inputs: btc.TransactionInput[] = decodeBitcoinTx(tx).inputs;
+
+        // Filter out intentional spend inscription txid from the check list
+        if (sendingInscriptionsList.length > 0) {
+          inputs = filterOutIntentionalInscriptionsSpend({
+            inputs: decodeBitcoinTx(tx).inputs,
+            inscriptions: sendingInscriptionsList,
+          });
+        }
```

Comment:
Anyway we could write this without reassigning the variable? Why not just use `filter`?

Reply:
yeah, agree here